geode-core/src/main/java/org/apache/geode/internal/cache/InternalCacheBuilder.java

```diff
@@ -102,20 +109,32 @@ public InternalCacheBuilder(CacheConfig cacheConfig) {
   }
 
   private InternalCacheBuilder(Properties configProperties, CacheConfig cacheConfig) {
-    this(configProperties, cacheConfig, InternalDistributedSystem::getConnectedInstance,
+    this(configProperties,
+        cacheConfig,
+        new CompositeMeterRegistryFactory() {},
+        compositeMeterRegistry -> CacheMetricsSession.builder().build(compositeMeterRegistry),
```

Comment:
Possibly could do `CacheMetricsSession.builder()::build` (though that syntax may scare people).